how-to-release.md

```diff
@@ -0,0 +1,298 @@
+
+```
+// Make sure you are on the right commit
+git checkout -b release-0.3.0-rc0
+```
+
+2. Remove the `-SNAPSHOT` suffix in all the versions of Pravega projects in `gradle.properties`. For example, if the current value is `pravegaVersion=0.3.0-SNAPSHOT`, then change it to `pravegaVersion=0.3.0`.
+
+3. Push this commit onto the release branch. The commit message should be `Add release X.Y.Z`. A pull request is NOT required for this commit.
+```
+git commit -s -am "Add release 0.3.0"
+git push origin release-0.3.0-rc0
+```
+
+4. Tag the commit with `vX.Y.Z-rcN` and push this tag. For example, `v0.3.0-rc0`.
+```
+git tag v0.3.0-rc0
+git push origin v0.3.0-rc0
+``` 
+
+5. Delete the temporary release branch.
+```
+git branch -D release-0.3.0-rc0
+```
+
+## Staging the binary releases
+
+Build the binaries with the following command, there could be some slight differences regarding the parameters and plugin used for publishing.
+```
+./gradlew clean assemble publish -PdoSigning=true -Psigning.password=<signing-password> -PpublishUrl=mavenCentral -PpublishUsername=<sonatype-username> -PpublishPassword=<sonatype-password>
+```
+
+* Login to [Nexus Repository Manager](https://oss.sonatype.org) using sonatype credentials with write access to io.pravega group.
+* Under Build Promotion -> Staging Repositories
+  * Drop the staging repository of the previous rejected release candidate if there is one
+  * Locate the staging repository that was created for the latest publish (format iopravega-XXXX, for example iopravega-1004)
+* Select the repository and click the "Close" button in the top menu bar. This will perform validations to ensure that the contents meets the maven requirements (contains signatures, javadocs, sources, etc). This operation takes a short time to complete, press the Refresh button in the top menu bar occasionally until the operation completes.
+* Once the operation completes, locate the URL field in the Summary tab of the newly closed repository (it will be something like https://oss.sonatype.org/content/repositories/iopravega-XXXX where XXXX is the number of the staging repository). This should be tested to ensure that all artifacts are present and functions as expected.
+
+## Staging the docker images
+
+The docker images should also be staged to the release manager's personal repository for validation from Docker/Kubernetes users. This can be done with the below command.
+```
+./gradlew clean docker dockerPush -PpravegaBaseTag=<docker-username>/pravega -PbookkeeperBaseTag=<docker-username>/bookkeeper -PpravegaVersion='X.Y.Z-rcN'
+```
+
+# Vote on the release candidate
+
+## Announcing for voting
+Once you have built and individually reviewed the release candidate, please share it for the community-wide review. You need to share these things to the dev mailing list for this release candidate and reference the link in the Slack channel.
```

Comment:
Once you have built and individually reviewed the release candidate, please share it for community review. The voting must happen in the dev mailing list (`cncf-pravega-dev@cncf.io`) for this release candidate and reference the link in the release Slack channel.